proknow-sdk/Requestor.cs

```diff
@@ -47,20 +59,43 @@ public Requestor(string baseUrl, string id, string secret)
         public async Task<string> DeleteAsync(string route, IList<KeyValuePair<string, string>> headerKeyValuePairs = null, HttpContent content = null)
         {
             var request = new HttpRequestMessage(HttpMethod.Delete, $"{_baseUrl}/{route}");
-            request.Headers.Authorization = _authenticationHeaderValue;
-            if (headerKeyValuePairs != null)
+            try
```

Comment:
After the request is constructed, the methods look like they're pretty similar to one another (maybe with the exception of the streaming one). Is there an opportunity to refactor this request building and error handling logic into a helper method?

Reply:
I will refactor common code into helper methods.